cli/ops/tls.rs

```diff
@@ -78,3 +107,101 @@ pub fn op_dial_tls(
 
   Ok(JsonOp::Async(Box::new(op)))
 }
+
+fn load_certs(path: &str) -> Result<Vec<Certificate>, ErrBox> {
+  let cert_file = File::open(path)?;
+  let reader = &mut BufReader::new(cert_file);
+  Ok(certs(reader).unwrap())
+}
+
+fn load_keys(path: &str) -> Result<Vec<PrivateKey>, ErrBox> {
+  let key_file = File::open(path)?;
+  let reader = &mut BufReader::new(key_file);
+  let keys = rsa_private_keys(reader).unwrap();
+  Ok(keys)
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct ListenTlsArgs {
+  transport: String,
+  hostname: String,
+  port: u16,
+  cert_file: String,
+  key_file: String,
+}
+
+fn op_listen_tls(
+  state: &ThreadSafeState,
+  args: Value,
+  _zero_copy: Option<PinnedBuf>,
+) -> Result<JsonOp, ErrBox> {
+  let args: ListenTlsArgs = serde_json::from_value(args)?;
+  assert_eq!(args.transport, "tcp");
+
+  // TODO(ry) Using format! is suboptimal here. Better would be if
+  // state.check_net and resolve_addr() took hostname and port directly.
+  let address = format!("{}:{}", args.hostname, args.port);
+
+  state.check_net(&address)?;
+
+  let mut config = ServerConfig::new(NoClientAuth::new());
+  config
+    .set_single_cert(
+      load_certs(&args.cert_file)?,
+      load_keys(&args.key_file)?.remove(0),
```

Comment:
Can you add a test for doing listenTLS with a missing cert / key path? Just check that we get a NotFound error - which I think should happen without modification.

Reply:
Thanks for pointing this out, I need to harden it a bit more to avoid runtime panics :+1: